server/src/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -962,18 +962,22 @@ protected void updateSite2SiteVpnConnectionState(final List<DomainRouterVO> rout
                         }
                         final Site2SiteVpnConnection.State oldState = conn.getState();
                         final Site2SiteCustomerGateway gw = _s2sCustomerGatewayDao.findById(conn.getCustomerGatewayId());
-                        if (answer.isConnected(gw.getGatewayIp())) {
-                            conn.setState(Site2SiteVpnConnection.State.Connected);
-                        } else {
-                            conn.setState(Site2SiteVpnConnection.State.Disconnected);
-                        }
-                        _s2sVpnConnectionDao.persist(conn);
-                        if (oldState != conn.getState()) {
-                            final String title = "Site-to-site Vpn Connection to " + gw.getName() + " just switch from " + oldState + " to " + conn.getState();
-                            final String context = "Site-to-site Vpn Connection to " + gw.getName() + " on router " + router.getHostName() + "(id: " + router.getId() + ") "
-                                    + " just switch from " + oldState + " to " + conn.getState();
-                            s_logger.info(context);
-                            _alertMgr.sendAlert(AlertManager.AlertType.ALERT_TYPE_DOMAIN_ROUTER, router.getDataCenterId(), router.getPodIdToDeployIn(), title, context);
+
+                        if (answer.isIPPresent(gw.getGatewayIp())) {
+                            if (answer.isConnected(gw.getGatewayIp())) {
+                                conn.setState(Site2SiteVpnConnection.State.Connected);
+                            } else {
+                                conn.setState(Site2SiteVpnConnection.State.Disconnected);
+                            }
+                            _s2sVpnConnectionDao.persist(conn);
+                            if (oldState != conn.getState()) {
+                                final String title = "Site-to-site Vpn Connection to " + gw.getName() + " just switch from " + oldState + " to " + conn.getState();
```

Comment:
Minor nit: could you please fix the grammatical error in this error message?  It should read "~~just~~ switch**ed** from".